### DISCUSSION TITLE
Fallback value

### DISCUSSION THREAD

**Discussion by JackOrez** [helgoboss/helgobox#1216] [Q&A]
I was thinking what would be nice is a definable fallback value, so that for example, when a source is done being tweaked after x amount of ms the target gets set to specified value, maybe even a user definable amount of ms for a linear transition to that fallback value.

**Comment by helgoboss** [comment]
Not 100% sure what you mean by that. I suppose you want to change values in using relative control and then always "snap back" to some sort of "center" value when you haven't touched the control element for a few milliseconds ... like a touch strip or pitch wheel...?

If yes, pretty sure you can already do that using a second mapping with control transformation. Have a look at the "debounce" control transformation preset and the "help" button, which should link to the technical reference.

Even linear transitions back to a value are possible with that.

**Reply by JackOrez** [reply]
Yes, you got the idea and your suggestion works well. I have another request if you don't mind :)

I was wondering if you could add the ability to import *.csv files into the 'Control transformation' editor?
I have some curves in this format that I would love to use in Helgobox. I have attached a random example.
This one is 512pts, but it would be nice to import ones of arbitrary length, maybe with an option to distribute 
the points over 'x' amount of time. 

Thank you :D

**Reply by helgoboss** [reply]
A quite specific thing which you could easily build yourself in Excel (or similar).

You just need to write some Excel formulas that generates something like this at the end:

```eel
y = x == 0 ? 0
  : x < 0.001957 ? 0.005180
  : x < 0.003914 ? 0.010390
  : x < 0.005871 ? 0.015630
  : 1;
```

Quite easily doable because you can have a very simple formula for each row that generates exactly one of those `: x < src ? dest` lines and then mark all cells and copy&paste.

**Comment by JackOrez** [comment]
[Curve.csv](https://github.com/user-attachments/files/17251446/Curve.csv)
